pkg/template/template.go

```diff
@@ -169,6 +169,10 @@ func (p *Processor) GenerateParameterValues(t *api.Template) error {
 				return fmt.Errorf("template.parameters[%v]: Unable to convert the generated value '%#v' to string", i, value)
 			}
 		}
+		if len(param.Value) == 0 && param.Required {
+			return fmt.Errorf("template.parameters[%v]: parameter is marked required but has no default value and no generator", i)
```

Comment:
That's probably true of all the parameter errors if we expect a client to be able to programmatically identify and present the error associated with the field